File: VelibStations_iOS/VelibStationsTests/Common/Database/DatabaseManagerUnitTest.swift
//
//  DatabaseManagerUnitTest.swift
//  VelibStationsTests
//
//  Created by Safi on 28/11/2019.
//

import XCTest
@testable import VelibStations
@testable import CommonKit

class DatabaseManagerUnitTest : UnitTest {
    
    override func setUp() {
        // Before every test.
        super.setUp()
    }

    override func tearDown() {
        // After every test.
        super.tearDown()
    }

    func testAccessVelibDatabase() {
        let databaseManager: IDatabaseManager = AddOnManager().getAddOn(type: AddOnType().DATABASE_MANAGER) as! IDatabaseManager
        let velibDatabase: IVelibDatabase? = databaseManager.getVelibDatabase() as IVelibDatabase
        XCTAssertTrue(velibDatabase != nil)
    }
}

File: VelibStations_Android/common/src/androidTest/kotlin/com/muhammedsafiulazam/common/database/DatabaseManagerUnitTest.kt
package com.muhammedsafiulazam.common.database

import com.muhammedsafiulazam.common.addon.AddOnManager
import com.muhammedsafiulazam.common.addon.AddOnType
import com.muhammedsafiulazam.common.core.AndroidUnitTest
import com.muhammedsafiulazam.common.database.velib.IVelibDatabase
import org.junit.Test
import kotlin.test.asserter

class DatabaseManagerUnitTest : AndroidUnitTest() {
    @Test
    /**
     * Access photo database.
     */
    fun accessVelibDatabase() {
        val databaseManager: IDatabaseManager = AddOnManager.getAddOn(AddOnType.DATABASE_MANAGER) as IDatabaseManager
        val velibDatabase: IVelibDatabase? = databaseManager.getVelibDatabase()
        asserter.assertTrue("accessVelibDatabase", velibDatabase != null)
    }
}